src/modules/collection/sagas.ts

```diff
@@ -226,3 +236,50 @@ function* handleMintColectionItems(action: MintCollectionItemsRequestAction) {
 function* handleConnectWalletSuccess() {
   yield put(fetchCollectionsRequest())
 }
+
+function* handleTransactionSuccess(action: FetchTransactionSuccessAction) {
+  const transaction = action.payload.transaction
+
+  try {
+    switch (transaction.actionType) {
+      case PUBLISH_COLLECTION_SUCCESS: {
+        // We re-fetch the collection from the store to get the updated version
+        const collectionId = transaction.payload.collection.id
+        const collection = yield select(state => getCollection(state, collectionId))
+        const items = yield select(state => getCollectionItems(state, collectionId))
+
+        for (const item of items) {
+          yield put(deployItemContentsRequest(collection, item))
+        }
+        break
+      }
+      default: {
+        break
+      }
+    }
+  } catch (error) {
+    console.error(error)
+  }
+}
+
+function* handleRequestCollectionSuccess(action: FetchCollectionsSuccessAction) {
+  const allItems: Item[] = yield select(getItems)
+  if (allItems.length === 0) {
+    yield take(FETCH_ITEMS_SUCCESS)
+  }
+
+  try {
+    const { collections } = action.payload
+
+    for (const collection of collections) {
+      if (!collection.isPublished) continue
+      for (const item of allItems) {
+        if (item.collectionId === collection.id) {
```

Comment:
here i'd add `&& !item.inCatalyst` so we prevent sending an unnecessary upload request

Reply:
Thought about it but wasn't sure, gotcha!